compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -428,6 +428,7 @@ pub enum RetagKind {
 
 /// The `FakeReadCause` describes the type of pattern why a FakeRead statement exists.
 #[derive(Copy, Clone, TyEncodable, TyDecodable, Debug, Hash, HashStable, PartialEq)]
+#[derive(TypeFoldable, TypeVisitable)]
```

Comment:
why is this needed

Reply:
`StatementKind::FakeRead` contains the tuple `(FakeReadCause, Place)`.  Since `Place` is not noop-traversable, neither is the tuple; and therefore the tuple (and `StatementKind`) is only traversable if `FakeReadCause` is.  Will add a comment to this effect.

Reply:
(This could be avoided by using a named struct in place of the tuple, upon which we `#[derive(TypeFoldable, TypeVisitable)]`).

Reply:
Replaced with named struct in b82e1d4b838cbf130b48b25d0f2031a4707e85f4